activesupport/lib/active_support/cache/memory_store.rb

```diff
@@ -114,45 +131,45 @@ def synchronize(&block) # :nodoc:
       private
         PER_ENTRY_OVERHEAD = 240
 
-        def cached_size(key, entry)
-          key.to_s.bytesize + entry.size + PER_ENTRY_OVERHEAD
+        def cached_size(key, payload)
+          key.to_s.bytesize + payload.bytesize + PER_ENTRY_OVERHEAD
         end
 
         def read_entry(key, **options)
           entry = nil
           synchronize do
-            entry = @data.delete(key)
-            if entry
-              @data[key] = entry
+            payload = @data.delete(key)
+            if payload
+              @data[key] = payload
+              entry = deserialize_entry(payload)
               entry = entry.dup
```

Comment:
One last thing: `DupCoder#load` has already copied the entry, so I don't think we need this line any more.

Reply:
Indeed. I just pushed that change.